### ISSUE TITLE
decode.js

### ISSUE THREAD

**Issue by hograthm** [alashow/datmusic-api#51]
Hello
I would be very grateful for an explanation of how this code works.
https://github.com/alashow/datmusic-api/blob/master/resources/js/decode.js#L2
[Formatted gist](https://gist.github.com/RichardVoid/ce8e3f7240f72b59cbd0537092967b6e)

What `r` and `t` mean in :
https://gist.github.com/RichardVoid/ce8e3f7240f72b59cbd0537092967b6e#L1
```javascript 
var decode = function(r, t)
```
And what is in `argv[2]`:
https://gist.github.com/RichardVoid/ce8e3f7240f72b59cbd0537092967b6e#L59
```javascript
process.stdout.write(decode(process.argv[2]));
```
Is there any plans to rewrite decode.js in PHP?

**Comment by alashow** [comment]
Hello,

See this [commits](https://github.com/alashow/datmusic-api/commit/2bc64323985f3a6a4d06c6abcca325f437ab00af) comment about this.

As the filename suggests, it decodes VK's encoded URLs.
The code itself was derived from VK's web client JavaScript ([here](https://vk.com/js/cmodules/web/audioplayer.js)).
Before 2bc64323985f3a6a4d06c6abcca325f437ab00af, VK was using different decoding code. When they started encoding their mp3 URL's I found PHP version of the code from some Russian forum. Now they have changed it, I don't want or even can't translate it to PHP. It's clearly written so other human beings couldn't understand. Even if I translated it, I would need to translate it again if they change it again.

See #34 about encoded mp3 URL's.

```javascript
var decode = function(r, t)
```
The `r` param is an encoded URL to decode. I'm not sure about `t`, I might have removed some parts of the decoder code. But it does the job.

[`argv[2]`](https://nodejs.org/docs/latest/api/process.html#process_process_argv) is to get a decoded URL passed from [PHP](https://github.com/alashow/datmusic-api/blob/96ed1d8e9b48459bb6bb2259b213277388a5c3a1/app/helpers.php#L36).
